### PR TITLE
Add split recording class and function

### PR DESCRIPTION
**Pull request by alejoe91**
New functions to split multi-segment recordings into a list of mono-segments:

- Return a list of mono-segment recordings
`recording_list = si.split_recording(rec_multi)`

- Return a single mono segment recording
`recording_mono = si.select_segment_recording(rec_multi, segment_index=1)`

### CODE REVIEW

File: spikeinterface/core/segmentutils.py [hunk 1]
```diff
@@ -218,6 +217,54 @@ def concatenate_recordings(*args, **kwargs):
...
+        assert segment_index < num_segments, f"'segment_index' must be between 0 and {num_segments - 1}"
+
+        rec_seg = recording._recording_segments[segment_index]
+        self.add_recording_segment(rec_seg)
+
+        self._kwargs = {'recording': recording.to_dict(),
+                        'segment_index': segment_index}
+        
+
+def split_recording(recording: BaseRecording):
```
Comment (samuelgarcia) [inline spikeinterface/core/segmentutils.py:247]: I not not sure about this name why not split_segment_recording.

File: spikeinterface/core/segmentutils.py [hunk 2]
```diff
@@ -218,6 +217,54 @@ def concatenate_recordings(*args, **kwargs):
 concatenate_recordings.__doc__ == ConcatenateSegmentRecording.__doc__
 
 
+class SplitSegmentRecording(BaseRecording):
```
Comment (samuelgarcia) [inline spikeinterface/core/segmentutils.py:220]: I think we should have somethin that select one or more segment and call this SelectSegmentRecroding.

than we could have 2 function on top of it : `split_segment` (or split_segment_recording) and `select_some_segment()`

Theses 2 could also be avialable directly with a method I think

Comment (alejoe91) [reply spikeinterface/core/segmentutils.py:220]: ok!

File: spikeinterface/core/segmentutils.py [hunk 3]
```diff
@@ -218,6 +221,63 @@ def concatenate_recordings(*args, **kwargs):
...
+        The segment indices to select
+    """
+
+    def __init__(self, recording: BaseRecording, segment_indices: List[int]):
+        BaseRecording.__init__(self, recording.get_sampling_frequency(), 
+                               recording.channel_ids, recording.get_dtype())
+        recording.copy_metadata(self)
+        
+        num_segments = recording.get_num_segments()
+        assert len(segment_indices) < num_segments, f"'segment_index' must be between 0 and {num_segments - 1}"
```
Comment (samuelgarcia) [inline spikeinterface/core/segmentutils.py:242]: assert all(0<= ind < num_segments for inf in segment_indices), ....

File: spikeinterface/core/segmentutils.py [hunk 4]
```diff
@@ -218,6 +221,63 @@ def concatenate_recordings(*args, **kwargs):
...
+
+        for segment_index in segment_indices:
+            rec_seg = recording._recording_segments[segment_index]
+            self.add_recording_segment(rec_seg)
+
+        self._kwargs = {'recording': recording.to_dict(),
+                        'segment_indices': segment_indices}
```
Comment (samuelgarcia) [inline spikeinterface/core/segmentutils.py:249]: You neeed to ensure that segment_indices is not a numpy array
segment_indices = segment_indices[int(e) for a in segment_indices]

File: spikeinterface/core/tests/test_segmentutils.py [hunk 5]
```diff
@@ -54,6 +55,27 @@ def test_append_concatenate_recordings():
...
 
+def test_split_recordings():
+    traces = np.zeros((1000, 5), dtype='float64')
+    traces[:] = np.arange(1000)[:, None]
+    sampling_frequency = 30000
+    rec0 = NumpyRecording([traces] * 3, sampling_frequency)
+    
+    rec_list = split_recording(rec0)
+    rec1 = select_segment_recording(rec0, segment_indices=[1])
+    rec2 = rec0.select_segments(segment_indices=[1, 2])
```
Comment (samuelgarcia) [inline spikeinterface/core/tests/test_segmentutils.py:66]: you should also test the case segment_indices is an integer

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**